crypto/pem/pem_lib.c

```diff
@@ -334,7 +334,7 @@ int PEM_ASN1_write_bio(i2d_of_void *i2d, const char *name, BIO *bp,
         }
     }
 
-    if ((dsize = i2d(x, NULL)) < 0) {
+    if ((dsize = i2d(x, NULL)) <= 0) {
```

Comment:
Notice the discussion at https://github.com/openssl/openssl/pull/12312#discussion_r446756676